aws/ecs/packer/to-upload/peers.sh

```diff
@@ -1,30 +1,235 @@
 #!/bin/bash
-# This script print a list of IPs of the EC2 instances in your current Autoscaling Group.
+# This script print a list of IPs based on
+#   1) TAG: this will match IPs for instances that have matching values for that Tag key set in /etc/weave/weave.config under
+#        the environment variable WEAVE_AWS_PEER_TAG
+#   2) Autoscaling Group: if this instance is part of an autoscaling group, this will return IPs for its peers
+#   3) Spot Fleet: Finally, if this instance is part of a spot fleet, this will return IPs for all instances in the fleet
 set -eu
 
-aws=/usr/local/bin/aws
+if [ -e '/etc/weave/weave.config' ]; then
+  source /etc/weave/weave.config
+fi
 
-export AWS_DEFAULT_REGION=$(curl -s curl http://169.254.169.254/latest/dynamic/instance-identity/document | \
-                            jq -r .region)
-current_instance_id=$(curl -s curl http://169.254.169.254/latest/meta-data/instance-id)
-current_autoscaling_group=$($aws autoscaling describe-auto-scaling-instances | \
-                            jq -r ".AutoScalingInstances[] | select(.InstanceId == \"$current_instance_id\") | .AutoScalingGroupName")
+WEAVE_AWS_PEER_TAG="${WEAVE_AWS_PEER_TAG:-}"
```

Comment:
You can do it like this https://github.com/weaveworks/integrations/blob/master/aws/ecs/packer/to-upload/run.sh#L28